ecs-agent/netlib/model/ecscni/eni_config.go

```diff
@@ -0,0 +1,89 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package ecscni
+
+import (
+	"fmt"
+
+	"github.com/aws/amazon-ecs-agent/ecs-agent/netlib/model/networkinterface"
+)
+
+// ENIConfig contains all the information needed to invoke the eni plugin
+type ENIConfig struct {
+	CNIConfig
+	// ENIID is the id of ec2 eni
+	ENIID string `json:"eni"`
```

Comment:
I have a feeling we can get rid of this field. I do not recall any of the CNI plugins using this, but I could be wrong.

Reply:
Will take a look

Reply:
Just checked, I think the model here is specifically for `vpc-eni` plugin and there is no reference to this field in this plugin; it uses MAC instead. So I think we are safe to remove it.

Reply:
Since we decided we are not renaming the ENIConfig object, I think this field will not look out of place. Thanks for taking a look.